### PR TITLE
Wizard Implementation

### PR DESCRIPTION
**Pull request by Camwyn**
Rework it all with hand-rolled tabs.
Also, convert all js to typescript - whee!

[TEC-5286]
[TEC-5287]
[TEC-5288]
[TEC-5289]
[TEC-5290]
[TEC-5291]
[TEC-5292]
[TEC-5293]

[TEC-5286]: https://stellarwp.atlassian.net/browse/TEC-5286?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TEC-5288]: https://stellarwp.atlassian.net/browse/TEC-5288?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TEC-5290]: https://stellarwp.atlassian.net/browse/TEC-5290?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TEC-5292]: https://stellarwp.atlassian.net/browse/TEC-5292?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[TEC-5287]: https://stellarwp.atlassian.net/browse/TEC-5287?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TEC-5289]: https://stellarwp.atlassian.net/browse/TEC-5289?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TEC-5291]: https://stellarwp.atlassian.net/browse/TEC-5291?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TEC-5293]: https://stellarwp.atlassian.net/browse/TEC-5293?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/resources/packages/wizard/components/tabs.tsx [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+	 * @property completed - Whether the tab is completed.
+	 * @property ref - A reference to the tab element.
+	 */
+	const InitialTabs = {
+		1: { id: "welcome", panelId: "welcomePanel", title: "Welcome", disabled: false, completed: false, ref: useRef(null) },
+		2: { id: "display", panelId: "displayPanel", title: "Display", disabled: true, completed: false, ref: useRef(null) },
+		3: { id: "settings", panelId: "settingsPanel", title: "Settings", disabled: true, completed: false, ref: useRef(null) },
+		4: { id: "organizer", panelId: "organizerPanel", title: "Organizer", disabled: true, completed: false, ref: useRef(null) },
+		5: { id: "venue", panelId: "venuePanel", title: "Venue", disabled: true, completed: false, ref: useRef(null) },
+		6: { id: "tickets", panelId: "ticketsPanel", title: "Tickets", disabled: true, completed: false, ref: useRef(null) }
```
Comment (bordoni) [inline src/resources/packages/wizard/components/tabs.tsx:36]: Translations.

File: src/resources/packages/wizard/components/tabs/display/tab.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+		'Week',
+		'Map',
+		'Photo',
+		'Summary',
+		'all'
+	];
+
+	document.addEventListener(
+		'change',
+		function(e) {
```
Comment (bordoni) [inline src/resources/packages/wizard/components/tabs/display/tab.tsx:21]:
```suggestion
	(event) => {
```

would also change from `e.` to `event.` in here.

File: src/resources/packages/wizard/components/tabs/tab.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+		<button
+		aria-controls={tab.panelId}
+		aria-selected={activeTab === index}
+		className="tec-events-onboarding__tab-button"
+		disabled={tab.disabled}
+		id={tab.id}
+		onClick={handleClick}
+		ref={tab.ref}
+		role="tab"
+		tabIndex={activeTab === index ? 0 : -1}
```
Comment (bordoni) [inline src/resources/packages/wizard/components/tabs/tab.tsx:55]:
```suggestion
			aria-controls={tab.panelId}
			aria-selected={activeTab === index}
			className="tec-events-onboarding__tab-button"
			disabled={tab.disabled}
			id={tab.id}
			onClick={handleClick}
			ref={tab.ref}
			role="tab"
			tabIndex={activeTab === index ? 0 : -1}
```

File: src/resources/packages/wizard/components/tabs/tickets/inputs/ticket-install.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+			<CheckboxControl
+				__nextHasNoMarginBottom
+				aria-describedby="tec-events-onboarding__checkbox-description"
+				checked={isChecked}
+				onChange={ setChecked }
+				id="tec-events-onboarding__tickets-checkbox-input"
+			/>
+			<div className="tec-events-onboarding__checkbox-description">
+				<label htmlFor="tec-events-onboarding__tickets-checkbox-input">
+				{__("Yes, install Event Tickets for free on my website.", "the-events-calendar")}
```
Comment (bordoni) [inline src/resources/packages/wizard/components/tabs/tickets/inputs/ticket-install.tsx:25]:
```suggestion
					{__("Yes, install Event Tickets for free on my website.", "the-events-calendar")}
```

File: src/resources/packages/wizard/components/tabs/welcome/inputs/opt-in.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+				id="tec-events-onboarding__optin-checkbox-input"
+			/>
+			<div className="tec-events-onboarding__checkbox-description">
+				<label htmlFor="tec-events-onboarding__optin-checkbox-input">
+				{__("Yes, I’d like to share basic information and have access to the TEC chatbot.", "the-events-calendar")}
+				</label>
+				<div
+					id="tec-events-onboarding__checkbox-description"
+				>
+				<a href="#" target="_blank">What permissions are being granted?</a>
```
Comment (bordoni) [inline src/resources/packages/wizard/components/tabs/welcome/inputs/opt-in.js:29]: Language.

**Review by bordoni — CHANGES_REQUESTED** [review]

File: package.json [hunk 6]
```diff
@@ -124,6 +124,8 @@
     "react": "^18.0.0",
     "react-day-picker": "^7.2.4",
     "react-dom": "^18.3.1",
+    "react-hook-form": "^7.43.9",
+    "react-router-dom": "^6.10.0",
```
Comment (Camwyn) [inline package.json:128]: Don't recall adding these?

File: .gitignore [hunk 7]
```diff
@@ -10,6 +10,9 @@ vendor
 !vendor/ForceUTF8
 !vendor/php-date-formatter
 
+# Build files - generated on packaging
+src/build
```
Comment (Camwyn) [inline .gitignore:14]: don't commit any build files!

**Pull request closed without merging**